### ISSUE TITLE
[DIY DNS Server] CodeGrade has internet settings set to "disabled"

### ISSUE THREAD

**Issue by gosh24** [atlarge-research/cn-lab-student#56]
As the title says. My server is working fine in the local environment, however, the print output (timeout when trying to connect to root DNS servers) suggests that this is the reason why tests on CodeGrade don't work.

gbv500

**Comment by gosh24** [comment]
@fzovpec Is that the case or is the issue with me?

**Comment by fzovpec** [comment]
Hey there! Are you using --ipv4only [https://computer-networks-lab.gitbook.io/computer-networks-lab-manual/assignments/dns](flag) in your implementation? There is an issue with a codegrade ipv6 support

**Comment by gosh24** [comment]
![image](https://github.com/atlarge-research/cn-lab-student/assets/90799368/8429f7ff-f14b-486e-92a0-d0af3f5f589b)

**Comment by gosh24** [comment]
Dear Gleb,

Thank you a lot for your prompt cooperation and enabling the settings. Unfortunately, there still seems to be a small issue before I can finally test once again. Error in CodeGrade is: 

The server output is /home/codegrade/.pyenv/versions/3.10.9/bin/python3: can't open file '/home/codegrade/student/../student/dns_check/dns.py': [Errno 2] No such file or directory

Perhaps the dns.py file needs to be copied in a different directory?

I really appreciate your help even outside lab hours.

**Comment by gosh24** [comment]
Perhaps:

mkdir $UPLOADED_FILES/../student/dns_check/
cp ./dns.py $UPLOADED_FILES/../student/dns_check/dns.py

**Comment by gosh24** [comment]
@fzovpec Thank you for being so understanding!

Alas, another peculiar issue has arose! All tests pass with "Success" now in the print output, but CodeGrade's final decision says tests fail with exit code 1. When I delved deeper into "check.py", I found that the TestCase.execute() method has a small mistake:

"return success" is indented to be only inside the except block, and the successful execution of the try block results in a "None" return. This in turn makes the final "if not execute_tests(..." method to evaluate to False and therefore give an exit code of 1.

gbv500
